### ISSUE TITLE
Update commits do not provide diffs

### ISSUE THREAD

**Issue by benkehoe** [PatMyron/cloudformation-resource-providers#3]
I guess the pull requests for updating the repo commits are automatically generated, and they use a commit that does not have the current head as a parent, so the `.gitmodules` file appears new every time. This means you can't see which repos have been updated. It'd be nice to have a meaningful diff (and history) to be able to know what's changing.
Example: https://github.com/PatMyron/cloudformation-resource-providers/pull/1/files#diff-fe7afb5c9c916e521401d3fcfb4277d5071798c3baf83baf11d6071742823584

**Comment by PatMyron** [comment]
Agreed that's better, here's where `commit-message` is set if anyone figures out improvements:
https://github.com/PatMyron/cloudformation-resource-providers/blob/main/.github/workflows/cron.yaml#L18-L22

**Comment by benkehoe** [comment]
Maybe adding `base: main`?

**Comment by PatMyron** [comment]
base branch for the automatic PR is already main:
https://github.com/PatMyron/cloudformation-resource-providers/pull/1

**Comment by benkehoe** [comment]
Yeah, the action notes that actions which checkout a commit may require `base: main` as input to the create-pull-request action. It's weird because the commit created has no parent

**Comment by benkehoe** [comment]
Wait. The default branch on the repo is `create-pull-request/patch`
![image](https://user-images.githubusercontent.com/1650723/135136351-d071a4da-7c7d-477c-b680-64b28425f60e.png)
and `main` is empty 🤔
![image](https://user-images.githubusercontent.com/1650723/135136432-c25b8343-3f5e-41de-91d5-914d594f7244.png)

**Comment by PatMyron** [comment]
work on `main` and the PR keeps force pushing a commit to `create-pull-request/patch` using `main` as base
(switched default to `create-pull-request/patch` so everyone can easily see the code written by those auto PRs)

**Comment by benkehoe** [comment]
this means that the diff of the PR is always against `main` which does not have `.gitmodules`. If the PR is never merged, `.gitmodules` is always new in the commit.

Wouldn't it make sense to have the action work on `main`, automatically push a commit (if anything has changed), and do work on workflows in a separate branch?

**Comment by PatMyron** [comment]
tbh haven't dug into git submodules enough to understand what happens when the same submodule is added again

---

agree the generic commit force-pushing without merging is hacky/unideal
fork, test, and PR if desired :)
(don't have much time on this now that I left CloudFormation)
